src/validation_mode.rs

```diff
@@ -0,0 +1,45 @@
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+/// The preferred mode of validation for struct (de)serialization.
+/// It also affects which format is used by the client when sending queries.
+///
+/// - [`ValidationMode::First`] enables validation _only for the first `N` rows_
+///   emitted by a cursor. For the following rows, validation is skipped.
+///   Format: `RowBinaryWithNamesAndTypes`.
+/// - [`ValidationMode::Each`] enables validation _for all rows_ emitted by a cursor.
+///   This is the slowest mode. Format: `RowBinaryWithNamesAndTypes`.
+///
+/// # Default
+///
+/// By default, [`ValidationMode::First`] with value `1` is used,
+/// meaning that only the first row will be validated against the database schema,
+/// which is extracted from the `RowBinaryWithNamesAndTypes` format header.
+/// It is done to minimize the performance impact of the validation,
+/// while still providing reasonable safety guarantees by default.
+///
+/// # Safety
```

Comment:
The "Safety" section in Rust has a special meaning, not applicable here

Reply:
It seems that the whole section is continuation of the previous one ("Default") describing only problems with that specific default value